build-tools/packages/build-tools/src/common/biomeConfig.ts

```diff
@@ -264,6 +243,64 @@ export class BiomeConfigReader {
 		const allConfigs = await getAllBiomeConfigPaths(configFile);
 		const mergedConfig = await loadBiomeConfigs(allConfigs);
 		const files = await getBiomeFormattedFiles(mergedConfig, directory, gitRepo);
-		return new BiomeConfigReader(configFile, allConfigs, mergedConfig, files);
+		return new BiomeConfigReaderV1(configFile, allConfigs, mergedConfig, files);
+	}
+}
+
+/**
```

Comment:
Nit: would it make sense to have split this module into code that is specifically for v1, and another that is for shared v1/v2 utilities? Or would that be more confusing?